x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -422,24 +422,35 @@ export class TaskRunner<
       );
     }
 
-    const maintenanceWindowIds = activeMaintenanceWindows
-      .filter(({ categoryIds }) => {
-        // If category IDs array doesn't exist: allow all
-        if (!Array.isArray(categoryIds)) {
-          return true;
-        }
-        // If category IDs array exist: check category
-        if ((categoryIds as string[]).includes(ruleType.category)) {
-          return true;
-        }
-        return false;
-      })
-      .map(({ id }) => id);
+    const maintenanceWindows = activeMaintenanceWindows.filter(({ categoryIds }) => {
+      // If category IDs array doesn't exist: allow all
+      if (!Array.isArray(categoryIds)) {
+        return true;
+      }
+      // If category IDs array exist: check category
+      if ((categoryIds as string[]).includes(ruleType.category)) {
+        return true;
+      }
+      return false;
+    });
+
+    const maintenanceWindowsIds = maintenanceWindows.map(({ id }) => id);
 
-    if (maintenanceWindowIds.length) {
-      this.alertingEventLogger.setMaintenanceWindowIds(maintenanceWindowIds);
+    const maintenanceWindowsWithScopedQuery = maintenanceWindows.filter(
+      ({ scopedQuery }) => scopedQuery
+    );
+    const maintenanceWindowsWithoutScopedQuery = maintenanceWindows.filter(
+      ({ scopedQuery }) => !scopedQuery
+    );
+
+    const maintenanceWindowsWithoutScopedQueryIds = maintenanceWindows.map(({ id }) => id);
+
+    if (maintenanceWindows.length) {
+      this.alertingEventLogger.setMaintenanceWindowIds(maintenanceWindowsIds);
```

Comment:
Since it's the rule's event log. The initial thinking was I would simply set all of the active MW ids during this execution. I suppose I should set only the Ids that affect some alert. In that case I will need to defer this call slightly.